### ISSUE TITLE
W0602 too limited in scope of analysis

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#177]
Originally reported by: **the mulhern (BitBucket: [the_mulhern](http://bitbucket.org/the_mulhern))**

---

It's possible to have a a file with something like:

```
#!python

var1 = {'a': 1}

def method1():
    global var1
    for k,v in var1:
        print("%s, %s" % (k,v))

def method2():
    global var1
    var1['b'] = 3
```

and so forth.

I think that there should not be an error message of the sort,
W0602(global-variable-not-assigned)...Using global for 'var1' but no assignment is done
in this case.

That is, analysis should detect the assignment in the file and not complain.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/177
